diff --git a/CMakeLists.txt b/CMakeLists.txt
index 18c4c2d..601ac94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,24 @@ set(MAJOR 1)
 set(MINOR 6)
 set(PATCH 0)
 
+# CMAKE_C_FLAGS and the like are _strings_, not lists.
+# So, we need a macro so that we can rewrite the values
+# in place, and avoid appending the flags twice
+macro(add_flags var flags)
+  string(FIND "${${var}}" ${flags} _res)
+  if(${_res} EQUAL "-1")
+    set(${var} "${${var}} ${flags}")
+  endif()
+endmacro()
+
+if (MINGW)
+  # optimization flags are missing by default, and cause "file too big" errors
+  add_flags(CMAKE_CXX_FLAGS_DEBUG "-Og")
+  add_flags(CMAKE_CXX_FLAGS_RELEASE "-O3")
+  add_flags(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g")
+  add_flags(CMAKE_CXX_FLAGS_MINSIZEREL "-Os")
+endif()
+
 if(NOT CMAKE_BUILD_TYPE )
 MESSAGE(STATUS "not define build type, set to release" )
 set(CMAKE_BUILD_TYPE Release )
